### DISCUSSION TITLE
Undefined constant "intval"

### DISCUSSION THREAD

**Discussion by adamesq** [YOURLS/YOURLS#3330] [Q&A]
I cannot figure out this one: Undefined constant "intval"

I tried updating. Followed directions and replaced files in the root, then even dug down and made sure updates were made. I updated PHP to 8.0 (it was too low). I then was prompted to update (Upgrade YOURLS Your current installation needs to be upgraded.), but Yourls is already showing 1.9. When I hit the "Update" button (which shouldn't be necessary since I manually updated, I get:

Fatal error: Uncaught Error: Undefined constant "intval" in <mydirectory>/admin/upgrade.php:32 Stack trace: #0 {main} thrown in <mydirectory>/admin/upgrade.php on line 32

**Comment by adamesq** [comment]
I removed '(interval)' from '$oldsql = (intval)($_GET['oldsql']);' in line 32 of the update.php file. The update "completed." I'm not sure what it actually did, but the install is working now.

I put the line back to '$oldsql = (intval)($_GET['oldsql']);' afterward, but I suspect there's an issue here.

**Reply by ozh** [reply]
Meh. My bad, probably a late night fix... It should be `intval($_GET['oldsql'])`, those extra parenthesis make no sense.

**Reply by ozh** [reply]
I made an issue to help people that would search for this -- https://github.com/YOURLS/YOURLS/issues/3331. Thanks for reporting the problem!

**Reply by adamesq** [reply]
I seem to have the displeasure of always running into these kinds of issues first 😂. Thank you so much for the quick follow-up!
